app/Nova/Actions/SyncEngagePurchaseRequestToQuickBooks.php

```diff
@@ -91,11 +90,11 @@ public function handle(ActionFields $fields, Collection $models)
         $item_ref = new IPPReferenceType();
         $item_ref->value = config('quickbooks.invoice.item_id');
 
-        $invoice->TxnDate = $envelope->submitted_at->format('Y/m/d');
-        $invoice->DueDate = $envelope->submitted_at->addDays(30)->format('Y/m/d');
+        $invoice->TxnDate = $engage_request->submitted_at->format('Y/m/d');
+        $invoice->DueDate = $engage_request->submitted_at->addDays(30)->format('Y/m/d');
         $invoice->CurrencyRef = $currency_ref;
-        $invoice->DocNumber = $envelope->id;
-        $invoice->PrivateNote = $reimburse_charge->PrivateNote.' | '.$envelope->description;
+        $invoice->DocNumber = $engage_request->engage_request_number;
```

Comment:
Conveniently, the highest DocuSign envelope number in prod is 829, and the first Engage request that needs to get pushed to QuickBooks is 1061.